File: 1-fundamentos-e-logica/9-exercicios-de-logica/Desafios/logica-pensamento-computacional/calculoSalario.js
// Desafios JavaScript tem funções "get" e "print" acessíveis globalmente:
//- "gets": Lê UMA linha com dado(s) de entrada (inputs) do usuário;
//- "print": imprime um texto de saída (output), pulando linha.

// Lê os valores de entrada:
const valorSalario = parseFloat(gets());
const valorBeneficios = parseFloat(gets());

// Calcula o imposto através da função "calcularImposto":
const valorImposto = calcularImposto(valorSalario);
// Calcula e imprime a saída (com 2 casas decimais):
const saida = valorSalario - valorImposto + valorBeneficios;
print(saida.toFixed(2));

// função útil para o calculo do imposto (baseado nas aliquotas):
function calcularImposto(salario) {
    let aliquota;
    if (salario >= 0 && salario <= 1100) {
        aliquota = 0.05;
    } else if ( salario >= 1100.1 && salario <= 2500) {
        aliquota = 0.10;
    } else {
        aliquota = 0.15;
    }

    return aliquota * salario;
    
}

File: 1-fundamentos-e-logica/9-exercicios-de-logica/Desafios/logica-pensamento-computacional/calculo_salario.py
# Para ler e escrever dados em Python, usamos as seguintes funções:
# input: lê UMA linha com dado(s) de entrada do usuário;
# print: imprime um texto de Saída (output), pulando linha;

# função útil para o calculo do imposto (baseado nas aliquotas):
def calcular_imposto(salario):
    aliquota = 0.0
    if (salario >= 0 and salario <= 1100):
        aliquota = 0.5
    elif (salario >= 1100.1 and salario <= 2500):
        aliquota = 0.10
    else:
        aliquota = 0.15
    
    return aliquota * salario

# Lê os valores de Entrada
valor_salario = float(input())
valor_beneficio = float(input())

#Calcula o imposto através da função calcular_imposto:
valor_imposto = calcular_imposto(valor_salario)
# Calcula e imprime a Saída (com 2 casas decimais):
saida = valor_salario - valor_imposto + valor_beneficio
print(f'{saida:.2f}')